File: 03-sum-nat/sum_nat.rb
# Ruby program to sum numbers from 1 to n
# Created by M.V.Harish Kumar on 10-09-2023
<<<<<<< HEAD
class Integer
    def sum_to(n)
    	(self..n).inject(:+)
    end
=======

def sum_to_n(n)
    (1..n).inject(:+)
>>>>>>> 1bc10791fb251e245e935b4186828b216d245e39
end

print "Enter value for `n`: "
n = gets.to_i
<<<<<<< HEAD
puts "The sum from 1 to #{n} is #{1.sum_to(n)}"
=======
puts "The sum from 1 to #{n} is #{sum_to_n n}"
>>>>>>> 1bc10791fb251e245e935b4186828b216d245e39

File: 03-sum-nat/sum_nat.go
// Go program to sum numbers from 1 to n
// Created by M.V.Harish Kumar on 25-10-2023
package main

import "fmt"

func sum_to_n(n uint) uint {
	var sum, i uint = 0, 0
	for ; i <= n; i++ {
		sum += i
	}
	return sum
}

func main() {
	var num uint
	fmt.Print("Enter value for `n`: ")
	fmt.Scanf("%d", &num)
	fmt.Printf("Sum of 1 to %d is %d\n", num, sum_to_n(num))
}
